crates/hir_ty/src/diagnostics/match_check.rs

```diff
@@ -767,10 +768,12 @@ impl Constructor {
 fn pat_constructor(cx: &MatchCheckCtx, pat: PatIdOrWild) -> MatchCheckResult<Option<Constructor>> {
     let res = match pat.as_pat(cx) {
         Pat::Wild => None,
-        // FIXME somehow create the Tuple constructor with the proper arity. If there are
-        // ellipsis, the arity is not equal to the number of patterns.
-        Pat::Tuple { args: pats, ellipsis } if ellipsis.is_none() => {
-            Some(Constructor::Tuple { arity: pats.len() })
+        Pat::Tuple { args: pats, ellipsis: None } => Some(Constructor::Tuple { arity: pats.len() }),
```

Comment:
I think we should probably just always do it like in the ellipsis case? :thinking:

Reply:
Right there isn't really a point in not doing that 👍